keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -222,25 +228,29 @@ Metadata:
   - cosi.io/finalizer [3]
 spec:
   bucketClassName: [4]
-  releasePolicy: {"Delete", "Retain"} [5]
-  bucketConfig: map[string]string [6]
-  bucketRef: [7]
+  supportedProtocols: [5]
+  releasePolicy: {"Delete", "Retain"} [6]
+  bucketConfig: map[string]string [7]
+  bucketRef: [8]
     name:
     namespace:
-  secretName: [8]
+  secretRef: [9]
+  accessMode: {"rw", "ro"} [10]
 status:
-  bucketMetaData: <map[string]string> [9]
-  phase: {"Bound", "Released", "Failed", "Errored"} [10]
+  bucketAttributes: <map[string]string> [11]
+  phase: {"Bound", "Released", "Failed", "Errored"} [12]
   conditions:
 ```
 1. `name`: Generated in the pattern of `“bucket-”<BUCKET-NAMESPACE>"-"<BUCKET-NAME>`
 1. `labels`: central controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc*.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
 1. `bucketClassName`: Name of the associated BucketClass
+1. `supportedProtocols`:  String array of protocols (e.g. s3, gcs, swift, etc.) supported by the associated object store.
 1. `releasePolicy`: the release policy defined in the associated BucketClass. (see [BucketClass](#BucketClass) for more information)
-1. `bucketConfig`: a string:string map of driver defined key-value pairs
+1. `bucketAttributes`: a string:string map of driver defined key-value pairs
 1. `bucketRef`: the name & namespace of the bound Bucket.
-1. `secretName`: the name of the sidecar-generated secret. It's namespace is assumed `cosi-system`.
+1. `secretRef`: the name of the sidecar-generated secret. It's namespace is assumed `cosi-system`.
+1. `accessMode`: The level of access granted to the credentials stored in `secretRef`, one of "read only" or "read/write".
```

Comment:
plural?

Reply:
I think it would be one or the other in the case of buckets.  It's not useful in brownfield/static cases as much as greenfield, where it would be an explicitly required mode.

Reply:
Is `accessMode` required? If not what is the default?

Reply:
I would think RW would be default as greenfield is going to be the most common use case.  In BF/static, we'll have to decide on the outcome of it conflicting with the BC.   That is: "should be overridden or fail"?  It may be that we drop this entirely from the Bucket and only define it in the BC.